{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {},
   "outputs": [],
   "source": [
    "import pandas as pd\n",
    "from scipy import stats\n",
    "from sklearn.metrics import roc_auc_score, matthews_corrcoef\n",
    "import numpy as np\n",
    "import os"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {},
   "outputs": [],
   "source": [
    "# This should be changed per assay\n",
    "base_dir='/n/groups/marks/projects/RNAgym/baselines/Nucleotide_Transformer/results/'\n",
    "model='Nucleotide Transformer'\n",
    "score_column = 'NT_scores'\n",
    "dataset_result_extension='.csv'"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 15,
   "metadata": {},
   "outputs": [],
   "source": [
    "# read in the reference wt sequences\n",
    "wt_seqs = pd.read_csv('/n/groups/marks/projects/RNAgym/mutational_assays/reference_sheet_cleaned_CAS.csv', encoding='latin-1')\n",
    "wt_seqs.dropna(inplace=True)\n",
    "wt_seqs['Year'] = wt_seqs['Year'].astype(int)\n",
    "wt_seqs['First Author Last Name'] = wt_seqs['First Author Last Name'].astype(str)\n",
    "wt_seqs['Molecule Type'] = wt_seqs['Molecule Type'].astype(str)\n",
    "wt_seqs.drop(['Unnamed: 0', 'index'], inplace=True, axis=1)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 16,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Single Mutations Metrics:\n",
      "               Type  Spearmans    AUC    MCC\n",
      "mRNA          mRNA      0.199  0.400 -0.138\n",
      "tRNA          tRNA      0.165  0.491  0.016\n",
      "Aptamer    Aptamer      0.105  0.438 -0.092\n",
      "Ribozyme  Ribozyme      0.126  0.504  0.019\n",
      "All            All      0.138  0.484 -0.009\n",
      "Multiple Mutations Metrics:\n",
      "               Type  Spearmans    AUC    MCC\n",
      "mRNA          mRNA      0.246  0.376 -0.177\n",
      "tRNA          tRNA      0.094  0.552  0.078\n",
      "Aptamer    Aptamer      0.145  0.463 -0.071\n",
      "Ribozyme  Ribozyme      0.107  0.488 -0.019\n",
      "All            All      0.124  0.481 -0.032\n",
      "Combined Mutations Metrics:\n",
      "               Type  Spearmans    AUC    MCC\n",
      "mRNA          mRNA      0.245  0.375 -0.177\n",
      "tRNA          tRNA      0.099  0.553  0.079\n",
      "Aptamer    Aptamer      0.159  0.454 -0.079\n",
      "Ribozyme  Ribozyme      0.105  0.488 -0.019\n",
      "All            All      0.126  0.479 -0.033\n"
     ]
    }
   ],
   "source": [
    "\n",
    "spearmans_single = []\n",
    "aucs_single = []\n",
    "mccs_single = []\n",
    "\n",
    "spearmans_multiple = []\n",
    "aucs_multiple = []\n",
    "mccs_multiple = []\n",
    "\n",
    "spearmans_combined = []\n",
    "aucs_combined = []\n",
    "mccs_combined = []\n",
    "\n",
    "for _, row in wt_seqs.iterrows():\n",
    "    dataset = row['fitness filename']\n",
    "    df_path = f'{base_dir}/{dataset}{dataset_result_extension}'\n",
    "    if os.path.exists(df_path):\n",
    "        df = pd.read_csv(df_path)\n",
    "        df.drop(['Unnamed: 0'], axis=1, inplace=True)\n",
    "#         df.drop(['Unnamed: 0'], axis=1, inplace=True)\n",
    "        assay_col = df.columns[1]\n",
    "        mutation_column = 'mutant' if 'mutant' in df.columns else 'mutation' if 'mutation' in df.columns else 'mutations' if 'mutations' in df.columns else None\n",
    "        \n",
    "        if mutation_column:\n",
    "            df['mutation_count'] = df[mutation_column].apply(lambda x: len(x.split(',')))\n",
    "            \n",
    "            single_mutation_df = df[df['mutation_count'] == 1]\n",
    "            multiple_mutation_df = df[df['mutation_count'] > 1]\n",
    "            \n",
    "            # Single mutations\n",
    "            if not single_mutation_df.empty:\n",
    "                assay_scores_single = np.array(single_mutation_df[assay_col])\n",
    "                model_stat_single = np.array(single_mutation_df[score_column])\n",
    "                \n",
    "                spearman_corr_single = stats.spearmanr(assay_scores_single, model_stat_single).correlation\n",
    "                spearmans_single.append(abs(spearman_corr_single))\n",
    "                \n",
    "                auc_single = roc_auc_score((assay_scores_single > np.median(assay_scores_single)).astype(int), model_stat_single)\n",
    "                aucs_single.append(auc_single)\n",
    "                \n",
    "                mcc_single = matthews_corrcoef((assay_scores_single > np.median(assay_scores_single)).astype(int), (model_stat_single > np.median(model_stat_single)).astype(int))\n",
    "                mccs_single.append(mcc_single)\n",
    "            else:\n",
    "                spearmans_single.append(np.nan)\n",
    "                aucs_single.append(np.nan)\n",
    "                mccs_single.append(np.nan)\n",
    "            \n",
    "            # Multiple mutations\n",
    "            if not multiple_mutation_df.empty:\n",
    "                assay_scores_multiple = np.array(multiple_mutation_df[assay_col])\n",
    "                model_stat_multiple = np.array(multiple_mutation_df[score_column])\n",
    "                \n",
    "                spearman_corr_multiple = stats.spearmanr(assay_scores_multiple, model_stat_multiple).correlation\n",
    "                spearmans_multiple.append(abs(spearman_corr_multiple))\n",
    "                \n",
    "                auc_multiple = roc_auc_score((assay_scores_multiple > np.median(assay_scores_multiple)).astype(int), model_stat_multiple)\n",
    "                aucs_multiple.append(auc_multiple)\n",
    "                \n",
    "                mcc_multiple = matthews_corrcoef((assay_scores_multiple > np.median(assay_scores_multiple)).astype(int), (model_stat_multiple > np.median(model_stat_multiple)).astype(int))\n",
    "                mccs_multiple.append(mcc_multiple)\n",
    "            else:\n",
    "                spearmans_multiple.append(np.nan)\n",
    "                aucs_multiple.append(np.nan)\n",
    "                mccs_multiple.append(np.nan)\n",
    "                \n",
    "            # Combined\n",
    "            assay_scores_combined = np.array(df[assay_col])\n",
    "            model_stat_combined = np.array(df[score_column])\n",
    "\n",
    "            spearman_corr_combined = stats.spearmanr(assay_scores_combined, model_stat_combined).correlation\n",
    "            spearmans_combined.append(abs(spearman_corr_combined))\n",
    "\n",
    "            auc_combined = roc_auc_score((assay_scores_combined > np.median(assay_scores_combined)).astype(int), model_stat_combined)\n",
    "            aucs_combined.append(auc_combined)\n",
    "\n",
    "            mcc_combined = matthews_corrcoef((assay_scores_combined > np.median(assay_scores_combined)).astype(int), (model_stat_combined > np.median(model_stat_combined)).astype(int))\n",
    "            mccs_combined.append(mcc_combined)\n",
    "        else:\n",
    "            spearmans_single.append(np.nan)\n",
    "            aucs_single.append(np.nan)\n",
    "            mccs_single.append(np.nan)\n",
    "            spearmans_multiple.append(np.nan)\n",
    "            aucs_multiple.append(np.nan)\n",
    "            mccs_multiple.append(np.nan)\n",
    "            spearmans_combined.append(np.nan)\n",
    "            aucs_combined.append(np.nan)\n",
    "            mccs_combined.append(np.nan)\n",
    "    else:\n",
    "        spearmans_single.append(np.nan)\n",
    "        aucs_single.append(np.nan)\n",
    "        mccs_single.append(np.nan)\n",
    "        spearmans_multiple.append(np.nan)\n",
    "        aucs_multiple.append(np.nan)\n",
    "        mccs_multiple.append(np.nan)\n",
    "        spearmans_combined.append(np.nan)\n",
    "        aucs_combined.append(np.nan)\n",
    "        mccs_combined.append(np.nan)\n",
    "\n",
    "# Group by type and calculate averages\n",
    "types = ['mRNA', 'tRNA', 'Aptamer', 'Ribozyme', 'All']\n",
    "wt_seqs['type'] = wt_seqs['fitness filename'].apply(lambda x: x.split('_')[-1])\n",
    "\n",
    "# Append new columns to wt_seqs for single and multiple mutations\n",
    "wt_seqs['Spearmans_single'] = spearmans_single\n",
    "wt_seqs['AUC_single'] = aucs_single\n",
    "wt_seqs['MCC_single'] = mccs_single\n",
    "\n",
    "wt_seqs['Spearmans_multiple'] = spearmans_multiple\n",
    "wt_seqs['AUC_multiple'] = aucs_multiple\n",
    "wt_seqs['MCC_multiple'] = mccs_multiple\n",
    "\n",
    "wt_seqs['Spearmans_combined'] = spearmans_combined\n",
    "wt_seqs['AUC_combined'] = aucs_combined\n",
    "wt_seqs['MCC_combined'] = mccs_combined\n",
    "\n",
    "# Calculate averages for each metric\n",
    "metrics = ['Spearmans', 'AUC', 'MCC']\n",
    "average_metrics_single = {}\n",
    "average_metrics_multiple = {}\n",
    "average_metrics_combined = {}\n",
    "\n",
    "for metric in metrics:\n",
    "    single_metric_col = f'{metric}_single'\n",
    "    multiple_metric_col = f'{metric}_multiple'\n",
    "    combined_metric_col = f'{metric}_combined'\n",
    "    \n",
    "    type_groups_single = wt_seqs.groupby('type')[single_metric_col].mean().reset_index()\n",
    "    type_groups_single.loc[len(type_groups_single.index)] = ['All', wt_seqs[single_metric_col].mean()]\n",
    "    averages_single = {type_: type_groups_single[type_groups_single['type'].str.contains(type_, case=False)][single_metric_col].mean() for type_ in types}\n",
    "    average_metrics_single[metric] = averages_single\n",
    "    \n",
    "    type_groups_multiple = wt_seqs.groupby('type')[multiple_metric_col].mean().reset_index()\n",
    "    type_groups_multiple.loc[len(type_groups_multiple.index)] = ['All', wt_seqs[multiple_metric_col].mean()]\n",
    "    averages_multiple = {type_: type_groups_multiple[type_groups_multiple['type'].str.contains(type_, case=False)][multiple_metric_col].mean() for type_ in types}\n",
    "    average_metrics_multiple[metric] = averages_multiple\n",
    "    \n",
    "    type_groups_combined = wt_seqs.groupby('type')[combined_metric_col].mean().reset_index()\n",
    "    type_groups_combined.loc[len(type_groups_combined.index)] = ['All', wt_seqs[combined_metric_col].mean()]\n",
    "    averages_combined = {type_: type_groups_combined[type_groups_combined['type'].str.contains(type_, case=False)][combined_metric_col].mean() for type_ in types}\n",
    "    average_metrics_combined[metric] = averages_combined\n",
    "\n",
    "# Create final result DataFrame\n",
    "result_df_single = pd.DataFrame(average_metrics_single)\n",
    "result_df_single.insert(0, 'Type', types)\n",
    "\n",
    "result_df_multiple = pd.DataFrame(average_metrics_multiple)\n",
    "result_df_multiple.insert(0, 'Type', types)\n",
    "\n",
    "result_df_combined = pd.DataFrame(average_metrics_combined)\n",
    "result_df_combined.insert(0, 'Type', types)\n",
    "\n",
    "with pd.option_context(\"display.precision\", 3):\n",
    "    print(\"Single Mutations Metrics:\\n\", result_df_single)\n",
    "    print(\"Multiple Mutations Metrics:\\n\", result_df_multiple)\n",
    "    print(\"Combined Mutations Metrics:\\n\", result_df_combined)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 17,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "/n/groups/marks/projects/RNAgym/baselines/Nucleotide_Transformer/results//ke_2017_mRNA.csv\n",
      "/n/groups/marks/projects/RNAgym/baselines/Nucleotide_Transformer/results//soo_2021_ribozyme.csv\n",
      "/n/groups/marks/projects/RNAgym/baselines/Nucleotide_Transformer/results//Andreasson_2020_ribozyme.csv\n",
      "/n/groups/marks/projects/RNAgym/baselines/Nucleotide_Transformer/results//Kobori_2015_tw_ribozyme.csv\n",
      "/n/groups/marks/projects/RNAgym/baselines/Nucleotide_Transformer/results//Kobori_2015_j12_ribozyme.csv\n",
      "/n/groups/marks/projects/RNAgym/baselines/Nucleotide_Transformer/results//Kobori_2015_p4_ribozyme.csv\n",
      "/n/groups/marks/projects/RNAgym/baselines/Nucleotide_Transformer/results//Guy_2014_tRNA.csv\n",
      "/n/groups/marks/projects/RNAgym/baselines/Nucleotide_Transformer/results//Kobori_2018_ribozyme.csv\n",
      "/n/groups/marks/projects/RNAgym/baselines/Nucleotide_Transformer/results//Townshend_2015_4nt_aptamer.csv\n",
      "/n/groups/marks/projects/RNAgym/baselines/Nucleotide_Transformer/results//Townshend_2015_5nt_aptamer.csv\n",
      "/n/groups/marks/projects/RNAgym/baselines/Nucleotide_Transformer/results//Townshend_2015_6nt_aptamer.csv\n",
      "/n/groups/marks/projects/RNAgym/baselines/Nucleotide_Transformer/results//Townshend_2015_7nt_aptamer.csv\n",
      "/n/groups/marks/projects/RNAgym/baselines/Nucleotide_Transformer/results//Townshend_2015_8nt_aptamer.csv\n",
      "/n/groups/marks/projects/RNAgym/baselines/Nucleotide_Transformer/results//Peri_2022_ribozyme.csv\n",
      "/n/groups/marks/projects/RNAgym/baselines/Nucleotide_Transformer/results//Roberts_2023_cepeb3_ribozyme.csv\n",
      "/n/groups/marks/projects/RNAgym/baselines/Nucleotide_Transformer/results//Roberts_2023_HDV_ribozyme.csv\n",
      "/n/groups/marks/projects/RNAgym/baselines/Nucleotide_Transformer/results//Roberts_2023_hp_ribozyme.csv\n",
      "/n/groups/marks/projects/RNAgym/baselines/Nucleotide_Transformer/results//Roberts_2023_hh_ribozyme.csv\n",
      "/n/groups/marks/projects/RNAgym/baselines/Nucleotide_Transformer/results//Roberts_2023_tw_ribozyme.csv\n",
      "/n/groups/marks/projects/RNAgym/baselines/Nucleotide_Transformer/results//Domingo_2018_tRNA.csv\n",
      "/n/groups/marks/projects/RNAgym/baselines/Nucleotide_Transformer/results//Janzen_2022_fam1a1_ribozyme.csv\n",
      "/n/groups/marks/projects/RNAgym/baselines/Nucleotide_Transformer/results//Janzen_2022_fam1b1_ribozyme.csv\n",
      "/n/groups/marks/projects/RNAgym/baselines/Nucleotide_Transformer/results//Janzen_2022_fam21_ribozyme.csv\n",
      "/n/groups/marks/projects/RNAgym/baselines/Nucleotide_Transformer/results//Janzen_2022_fam22_ribozyme.csv\n",
      "/n/groups/marks/projects/RNAgym/baselines/Nucleotide_Transformer/results//Janzen_2022_fam31_ribozyme.csv\n",
      "/n/groups/marks/projects/RNAgym/baselines/Nucleotide_Transformer/results//Beck_2022_ribozyme.csv\n",
      "/n/groups/marks/projects/RNAgym/baselines/Nucleotide_Transformer/results//Li_2016_tRNA.csv\n",
      "/n/groups/marks/projects/RNAgym/baselines/Nucleotide_Transformer/results//Tome_2014_GFP_aptamer.csv\n",
      "/n/groups/marks/projects/RNAgym/baselines/Nucleotide_Transformer/results//Tome_2014_NELFE_aptamer.csv\n",
      "/n/groups/marks/projects/RNAgym/baselines/Nucleotide_Transformer/results//Julien_2016_mRNA.csv\n"
     ]
    }
   ],
   "source": [
    "# read in the reference wt sequences\n",
    "wt_seqs = pd.read_csv('/n/groups/marks/projects/RNAgym/mutational_assays/reference_sheet_cleaned_CAS.csv', encoding='latin-1')\n",
    "wt_seqs.dropna(inplace=True)\n",
    "wt_seqs['Year'] = wt_seqs['Year'].astype(int)\n",
    "wt_seqs['First Author Last Name'] = wt_seqs['First Author Last Name'].astype(str)\n",
    "wt_seqs['Molecule Type'] = wt_seqs['Molecule Type'].astype(str)\n",
    "wt_seqs.drop(['Unnamed: 0', 'index'], inplace=True, axis=1)\n",
    "\n",
    "spearmans = []\n",
    "aucs = []\n",
    "mccs = []\n",
    "\n",
    "for _, row in wt_seqs.iterrows():\n",
    "    dataset = row['fitness filename']\n",
    "    df_path = f'{base_dir}/{dataset}{dataset_result_extension}'\n",
    "    print(df_path)\n",
    "    if os.path.exists(df_path):\n",
    "        df = pd.read_csv(df_path)\n",
    "        df.drop(['Unnamed: 0'], axis=1, inplace=True)\n",
    "        assay_col = df.columns[1]\n",
    "        \n",
    "        assay_scores = np.array(df[assay_col])\n",
    "        model_stat = np.array(df[score_column])\n",
    "        \n",
    "        # Spearman correlation\n",
    "        spearman_corr = stats.spearmanr(assay_scores, model_stat).correlation\n",
    "        spearmans.append(abs(spearman_corr))\n",
    "\n",
    "        # AUC\n",
    "        auc = roc_auc_score((assay_scores > np.median(assay_scores)).astype(int), model_stat)\n",
    "        aucs.append(auc)\n",
    "\n",
    "        # MCC\n",
    "        mcc = matthews_corrcoef((assay_scores > np.median(assay_scores)).astype(int), (model_stat > np.median(model_stat)).astype(int))\n",
    "        mccs.append(mcc)\n",
    "    else:\n",
    "        spearmans.append(np.nan)\n",
    "        aucs.append(np.nan)\n",
    "        mccs.append(np.nan)\n",
    "\n",
    "# Append new columns to wt_seqs\n",
    "wt_seqs['Spearman'] = spearmans\n",
    "wt_seqs['AUC'] = aucs\n",
    "wt_seqs['MCC'] = mccs\n",
    "\n",
    "# Calculate averages for each metric\n",
    "# metrics = ['Spearman', 'AUC', 'MCC']\n",
    "# average_metrics = {}\n",
    "\n",
    "# for metric in metrics:\n",
    "#     type_groups = wt_seqs.groupby('type')[metric].mean().reset_index()\n",
    "#     type_groups.loc[len(type_groups.index)] = ['All', wt_seqs[metric].mean()]\n",
    "#     averages = {type_: type_groups[type_groups['type'].str.contains(type_, case=False)][metric].mean() for type_ in types}\n",
    "#     average_metrics[metric] = averages\n",
    "\n",
    "# # Create final result DataFrame\n",
    "# result_df = pd.DataFrame(average_metrics)\n",
    "# result_df.insert(0, 'Type', types)\n",
    "\n",
    "# with pd.option_context(\"display.precision\", 3):\n",
    "#     print(\"Assay Metrics:\\n\", result_df)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 18,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Title</th>\n",
       "      <th>First Author Last Name</th>\n",
       "      <th>Year</th>\n",
       "      <th>Molecule Type</th>\n",
       "      <th>Raw Construct Sequence</th>\n",
       "      <th>fitness filename</th>\n",
       "      <th>Spearman</th>\n",
       "      <th>AUC</th>\n",
       "      <th>MCC</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>Saturation mutagenesis reveals manifold determ...</td>\n",
       "      <td>Ke</td>\n",
       "      <td>2017</td>\n",
       "      <td>mRNA</td>\n",
       "      <td>ccccacctcttcttcttttctagAGTTGCTGCTGGGAGCTCCAGCA...</td>\n",
       "      <td>ke_2017_mRNA</td>\n",
       "      <td>0.383205</td>\n",
       "      <td>0.303543</td>\n",
       "      <td>-0.279279</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>Fitness landscape of a dynamic RNA structure</td>\n",
       "      <td>Soo</td>\n",
       "      <td>2021</td>\n",
       "      <td>ribozyme</td>\n",
       "      <td>CTCTCTAGGTAGCAATATTACGACATGGAGGGAAAAGTTATCAGGC...</td>\n",
       "      <td>soo_2021_ribozyme</td>\n",
       "      <td>0.009862</td>\n",
       "      <td>0.496602</td>\n",
       "      <td>-0.008671</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>Comprehensive sequence-to-function mapping of ...</td>\n",
       "      <td>Andreasson</td>\n",
       "      <td>2020</td>\n",
       "      <td>ribozyme riboswitch</td>\n",
       "      <td>ATAAAAGCGCCAGAACTACAGAAATGTAGTTGACGAGGAGGAGGTT...</td>\n",
       "      <td>Andreasson_2020_ribozyme</td>\n",
       "      <td>0.114690</td>\n",
       "      <td>0.447871</td>\n",
       "      <td>-0.065024</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>High-throughput assay and engineering of self-...</td>\n",
       "      <td>Kobori</td>\n",
       "      <td>2015</td>\n",
       "      <td>Lib-Tw</td>\n",
       "      <td>TAATACGACTCACTATAGGGCGCGGCATTAATGCAGCTTTATTGGA...</td>\n",
       "      <td>Kobori_2015_tw_ribozyme</td>\n",
       "      <td>0.018681</td>\n",
       "      <td>0.495067</td>\n",
       "      <td>-0.007264</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>High-throughput assay and engineering of self-...</td>\n",
       "      <td>Kobori</td>\n",
       "      <td>2015</td>\n",
       "      <td>Lib-J1/2</td>\n",
       "      <td>TAATACGACTCACTATAGGGCCGCGACTCTAGAAGTGATGCTCTGC...</td>\n",
       "      <td>Kobori_2015_j12_ribozyme</td>\n",
       "      <td>0.229690</td>\n",
       "      <td>0.620117</td>\n",
       "      <td>0.140625</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>High-throughput assay and engineering of self-...</td>\n",
       "      <td>Kobori</td>\n",
       "      <td>2015</td>\n",
       "      <td>Lib-P4</td>\n",
       "      <td>TAATACGACTCACTATAGGGCCGCGACTCTAGAAGTGATGCTCTGC...</td>\n",
       "      <td>Kobori_2015_p4_ribozyme</td>\n",
       "      <td>0.004775</td>\n",
       "      <td>0.468445</td>\n",
       "      <td>-0.046875</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>Identification of the determinants of tRNA fun...</td>\n",
       "      <td>Guy</td>\n",
       "      <td>2014</td>\n",
       "      <td>tRNA</td>\n",
       "      <td>GATCTAACAAAGTTCATAAAGAAATTACTCTCGGTAGCCAAGTTGG...</td>\n",
       "      <td>Guy_2014_tRNA</td>\n",
       "      <td>0.046266</td>\n",
       "      <td>0.533315</td>\n",
       "      <td>0.064516</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>Deep sequencing analysis of aptazyme variants ...</td>\n",
       "      <td>Kobori</td>\n",
       "      <td>2018</td>\n",
       "      <td>pistol ribozyme</td>\n",
       "      <td>TAATACGACTCACTATAGGGTCGNNNNNNNATAATCGCGTGGATAT...</td>\n",
       "      <td>Kobori_2018_ribozyme</td>\n",
       "      <td>0.084779</td>\n",
       "      <td>0.459204</td>\n",
       "      <td>-0.052490</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>High-throughput cellular RNA device engineering</td>\n",
       "      <td>Townshend</td>\n",
       "      <td>2015</td>\n",
       "      <td>4nt RNA aptamer</td>\n",
       "      <td>GCTGTCACCGGANNNNNNNNNTCNNNTNNNTNCNNNNNNNNNTCNN...</td>\n",
       "      <td>Townshend_2015_4nt_aptamer</td>\n",
       "      <td>0.093221</td>\n",
       "      <td>0.533514</td>\n",
       "      <td>-0.004854</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>High-throughput cellular RNA device engineering</td>\n",
       "      <td>Townshend</td>\n",
       "      <td>2015</td>\n",
       "      <td>5nt RNA aptamer</td>\n",
       "      <td>GCTGTCACCGGANNNNNNNCNNNNTNNTGANNCCNTNNNNNNNNCG...</td>\n",
       "      <td>Townshend_2015_5nt_aptamer</td>\n",
       "      <td>0.220651</td>\n",
       "      <td>0.374713</td>\n",
       "      <td>-0.200563</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10</th>\n",
       "      <td>High-throughput cellular RNA device engineering</td>\n",
       "      <td>Townshend</td>\n",
       "      <td>2015</td>\n",
       "      <td>6nt RNA aptamer</td>\n",
       "      <td>GCTGTCACCGGANNNNNNNCNNNNNNNNNNNGNCCNTNNNAGNNNN...</td>\n",
       "      <td>Townshend_2015_6nt_aptamer</td>\n",
       "      <td>0.208933</td>\n",
       "      <td>0.586520</td>\n",
       "      <td>0.113537</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11</th>\n",
       "      <td>High-throughput cellular RNA device engineering</td>\n",
       "      <td>Townshend</td>\n",
       "      <td>2015</td>\n",
       "      <td>7nt RNA aptamer</td>\n",
       "      <td>GCTGTCACCGGANNNNNNNNNNNGTCTNNNNNNNCNNNNNNNNCNN...</td>\n",
       "      <td>Townshend_2015_7nt_aptamer</td>\n",
       "      <td>0.037624</td>\n",
       "      <td>0.476610</td>\n",
       "      <td>-0.037254</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12</th>\n",
       "      <td>High-throughput cellular RNA device engineering</td>\n",
       "      <td>Townshend</td>\n",
       "      <td>2015</td>\n",
       "      <td>8nt RNA aptamer</td>\n",
       "      <td>GCTGTCACCGGANNNNNNNNNNCGNNNTGATGNNNNNNNNNNNNCN...</td>\n",
       "      <td>Townshend_2015_8nt_aptamer</td>\n",
       "      <td>0.226219</td>\n",
       "      <td>0.385770</td>\n",
       "      <td>-0.163823</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13</th>\n",
       "      <td>Dynamic RNA Fitness Landscapes of a Group I Ri...</td>\n",
       "      <td>Peri</td>\n",
       "      <td>2022</td>\n",
       "      <td>Azoarcus group I ribozyme</td>\n",
       "      <td>GUGCCUUGCGCCGGGAAACCACGCAAGGGAUGGUGUCAAAUUCGGC...</td>\n",
       "      <td>Peri_2022_ribozyme</td>\n",
       "      <td>0.026147</td>\n",
       "      <td>0.500842</td>\n",
       "      <td>-0.008057</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>14</th>\n",
       "      <td>RNA sequence to structure analysis from compre...</td>\n",
       "      <td>Roberts</td>\n",
       "      <td>2023</td>\n",
       "      <td>CPEB3 Ribozyme</td>\n",
       "      <td>GGGGGCCACAGCAGAAGCGTTCACGTCGCGGCCCCTGTCAGATTCT...</td>\n",
       "      <td>Roberts_2023_cepeb3_ribozyme</td>\n",
       "      <td>0.219021</td>\n",
       "      <td>0.582899</td>\n",
       "      <td>0.117490</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15</th>\n",
       "      <td>RNA sequence to structure analysis from compre...</td>\n",
       "      <td>Roberts</td>\n",
       "      <td>2023</td>\n",
       "      <td>HDV Ribozyme</td>\n",
       "      <td>GGCCGGCATGGTCCCAGCCTCCTCGCTGGCGCCGGCTGGGCAACAT...</td>\n",
       "      <td>Roberts_2023_HDV_ribozyme</td>\n",
       "      <td>0.194930</td>\n",
       "      <td>0.405339</td>\n",
       "      <td>-0.130178</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>16</th>\n",
       "      <td>RNA sequence to structure analysis from compre...</td>\n",
       "      <td>Roberts</td>\n",
       "      <td>2023</td>\n",
       "      <td>Hairpin Ribozyme</td>\n",
       "      <td>GTCCTCTTTCACGTGCGCGTGAAAGAGAGAAGTGAACCAGAGAAAC...</td>\n",
       "      <td>Roberts_2023_hp_ribozyme</td>\n",
       "      <td>0.053403</td>\n",
       "      <td>0.490231</td>\n",
       "      <td>-0.008770</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>17</th>\n",
       "      <td>RNA sequence to structure analysis from compre...</td>\n",
       "      <td>Roberts</td>\n",
       "      <td>2023</td>\n",
       "      <td>Hammerhead Ribozyme</td>\n",
       "      <td>ACCGGATGTGCTTTCCGGTCTGATGAGTCCGTGAGGACGAAACAG</td>\n",
       "      <td>Roberts_2023_hh_ribozyme</td>\n",
       "      <td>0.043122</td>\n",
       "      <td>0.474383</td>\n",
       "      <td>-0.029961</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>18</th>\n",
       "      <td>RNA sequence to structure analysis from compre...</td>\n",
       "      <td>Roberts</td>\n",
       "      <td>2023</td>\n",
       "      <td>Twister Ribozyme</td>\n",
       "      <td>AACACTGCCAATGCCGGTCCCAAGCCCGGATAAAAGTGGAGGGGGCGG</td>\n",
       "      <td>Roberts_2023_tw_ribozyme</td>\n",
       "      <td>0.019867</td>\n",
       "      <td>0.489497</td>\n",
       "      <td>-0.023699</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>19</th>\n",
       "      <td>Pairwise and higher-order genetic interactions...</td>\n",
       "      <td>Domingo</td>\n",
       "      <td>2018</td>\n",
       "      <td>tRNA</td>\n",
       "      <td>GUUCCGUUGGCGUAAUGGUAACGCGUCUCCCUCCUAAGGAGAAGAC...</td>\n",
       "      <td>Domingo_2018_tRNA</td>\n",
       "      <td>0.190741</td>\n",
       "      <td>0.597690</td>\n",
       "      <td>0.131018</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>20</th>\n",
       "      <td>Emergent properties as by-products of prebioti...</td>\n",
       "      <td>Janzen</td>\n",
       "      <td>2022</td>\n",
       "      <td>Fam1A.1 aminoacylation ribozyme</td>\n",
       "      <td>GATAATACGACTCACTATAGGGAATGGATCCACATCTACGAATTCC...</td>\n",
       "      <td>Janzen_2022_fam1a1_ribozyme</td>\n",
       "      <td>0.082191</td>\n",
       "      <td>0.540690</td>\n",
       "      <td>0.055811</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>21</th>\n",
       "      <td>Emergent properties as by-products of prebioti...</td>\n",
       "      <td>Janzen</td>\n",
       "      <td>2022</td>\n",
       "      <td>Fam1B.1 aminoacylation ribozyme</td>\n",
       "      <td>GATAATACGACTCACTATAGGGAATGGATCCACATCTACGAATTCC...</td>\n",
       "      <td>Janzen_2022_fam1b1_ribozyme</td>\n",
       "      <td>0.038526</td>\n",
       "      <td>0.531691</td>\n",
       "      <td>0.047619</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>22</th>\n",
       "      <td>Emergent properties as by-products of prebioti...</td>\n",
       "      <td>Janzen</td>\n",
       "      <td>2022</td>\n",
       "      <td>Fam2.1 aminoacylation ribozyme</td>\n",
       "      <td>GATAATACGACTCACTATAGGGAATGGATCCACATCTACGAATTCA...</td>\n",
       "      <td>Janzen_2022_fam21_ribozyme</td>\n",
       "      <td>0.264427</td>\n",
       "      <td>0.359427</td>\n",
       "      <td>-0.208398</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>23</th>\n",
       "      <td>Emergent properties as by-products of prebioti...</td>\n",
       "      <td>Janzen</td>\n",
       "      <td>2022</td>\n",
       "      <td>Fam2.2 aminoacylation ribozyme</td>\n",
       "      <td>GATAATACGACTCACTATAGGGAATGGATCCACATCTACGAATTCA...</td>\n",
       "      <td>Janzen_2022_fam22_ribozyme</td>\n",
       "      <td>0.007434</td>\n",
       "      <td>0.503460</td>\n",
       "      <td>0.008704</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>24</th>\n",
       "      <td>Emergent properties as by-products of prebioti...</td>\n",
       "      <td>Janzen</td>\n",
       "      <td>2022</td>\n",
       "      <td>Fam3.1 aminoacylation ribozyme</td>\n",
       "      <td>GATAATACGACTCACTATAGGGAATGGATCCACATCTACGAATTCA...</td>\n",
       "      <td>Janzen_2022_fam31_ribozyme</td>\n",
       "      <td>0.278336</td>\n",
       "      <td>0.345498</td>\n",
       "      <td>-0.224783</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>25</th>\n",
       "      <td>Predicting higher-order mutational effects in ...</td>\n",
       "      <td>Beck</td>\n",
       "      <td>2022</td>\n",
       "      <td>ribozyme</td>\n",
       "      <td>GGGGGCCACAGCAGAAGCGTTCACGTCGCGGCCCCTGTCAGATTCT...</td>\n",
       "      <td>Beck_2022_ribozyme</td>\n",
       "      <td>0.206225</td>\n",
       "      <td>0.577197</td>\n",
       "      <td>0.110454</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>26</th>\n",
       "      <td>The fitness landscape of a tRNA gene</td>\n",
       "      <td>Li</td>\n",
       "      <td>2016</td>\n",
       "      <td>tRNA</td>\n",
       "      <td>GTTCCGTTGGCGTAATGGTAACGCGTCTCCCTCCTAAGGAGAAGAC...</td>\n",
       "      <td>Li_2016_tRNA</td>\n",
       "      <td>0.060912</td>\n",
       "      <td>0.529479</td>\n",
       "      <td>0.042133</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>27</th>\n",
       "      <td>Comprehensive analysis of RNA-protein interact...</td>\n",
       "      <td>Tome</td>\n",
       "      <td>2014</td>\n",
       "      <td>GFP aptamer</td>\n",
       "      <td>GCUUCUGGACUGCGAUGGGAGCACGAAACGUCGUGGCGCAAUUGGG...</td>\n",
       "      <td>Tome_2014_GFP_aptamer</td>\n",
       "      <td>0.282189</td>\n",
       "      <td>0.363751</td>\n",
       "      <td>-0.189455</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>28</th>\n",
       "      <td>Comprehensive analysis of RNA-protein interact...</td>\n",
       "      <td>Tome</td>\n",
       "      <td>2014</td>\n",
       "      <td>NELF-E aptamer</td>\n",
       "      <td>CCCAACGACUGCCGAGCGAGAUUACGCUUGAGCGCCCCACUGAGGA...</td>\n",
       "      <td>Tome_2014_NELFE_aptamer</td>\n",
       "      <td>0.041492</td>\n",
       "      <td>0.457346</td>\n",
       "      <td>-0.070890</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>29</th>\n",
       "      <td>The complete local genotypeÃÂphenotype lands...</td>\n",
       "      <td>Julien</td>\n",
       "      <td>2016</td>\n",
       "      <td>mRNA</td>\n",
       "      <td>GATCCAGATCTAACTTGGGGTGGCTTTGTCTTCTTCTTTTGCCAAT...</td>\n",
       "      <td>Julien_2016_mRNA</td>\n",
       "      <td>0.105890</td>\n",
       "      <td>0.447097</td>\n",
       "      <td>-0.075486</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                                                Title First Author Last Name  \\\n",
       "0   Saturation mutagenesis reveals manifold determ...                     Ke   \n",
       "1        Fitness landscape of a dynamic RNA structure                    Soo   \n",
       "2   Comprehensive sequence-to-function mapping of ...             Andreasson   \n",
       "3   High-throughput assay and engineering of self-...                 Kobori   \n",
       "4   High-throughput assay and engineering of self-...                 Kobori   \n",
       "5   High-throughput assay and engineering of self-...                 Kobori   \n",
       "6   Identification of the determinants of tRNA fun...                    Guy   \n",
       "7   Deep sequencing analysis of aptazyme variants ...                 Kobori   \n",
       "8     High-throughput cellular RNA device engineering              Townshend   \n",
       "9     High-throughput cellular RNA device engineering              Townshend   \n",
       "10    High-throughput cellular RNA device engineering              Townshend   \n",
       "11    High-throughput cellular RNA device engineering              Townshend   \n",
       "12    High-throughput cellular RNA device engineering              Townshend   \n",
       "13  Dynamic RNA Fitness Landscapes of a Group I Ri...                   Peri   \n",
       "14  RNA sequence to structure analysis from compre...                Roberts   \n",
       "15  RNA sequence to structure analysis from compre...                Roberts   \n",
       "16  RNA sequence to structure analysis from compre...                Roberts   \n",
       "17  RNA sequence to structure analysis from compre...                Roberts   \n",
       "18  RNA sequence to structure analysis from compre...                Roberts   \n",
       "19  Pairwise and higher-order genetic interactions...                Domingo   \n",
       "20  Emergent properties as by-products of prebioti...                 Janzen   \n",
       "21  Emergent properties as by-products of prebioti...                 Janzen   \n",
       "22  Emergent properties as by-products of prebioti...                 Janzen   \n",
       "23  Emergent properties as by-products of prebioti...                 Janzen   \n",
       "24  Emergent properties as by-products of prebioti...                 Janzen   \n",
       "25  Predicting higher-order mutational effects in ...                   Beck   \n",
       "26               The fitness landscape of a tRNA gene                     Li   \n",
       "27  Comprehensive analysis of RNA-protein interact...                   Tome   \n",
       "28  Comprehensive analysis of RNA-protein interact...                   Tome   \n",
       "29  The complete local genotypeÃÂphenotype lands...                 Julien   \n",
       "\n",
       "    Year                    Molecule Type  \\\n",
       "0   2017                             mRNA   \n",
       "1   2021                        ribozyme    \n",
       "2   2020              ribozyme riboswitch   \n",
       "3   2015                           Lib-Tw   \n",
       "4   2015                         Lib-J1/2   \n",
       "5   2015                           Lib-P4   \n",
       "6   2014                             tRNA   \n",
       "7   2018                  pistol ribozyme   \n",
       "8   2015                  4nt RNA aptamer   \n",
       "9   2015                  5nt RNA aptamer   \n",
       "10  2015                  6nt RNA aptamer   \n",
       "11  2015                  7nt RNA aptamer   \n",
       "12  2015                  8nt RNA aptamer   \n",
       "13  2022        Azoarcus group I ribozyme   \n",
       "14  2023                   CPEB3 Ribozyme   \n",
       "15  2023                     HDV Ribozyme   \n",
       "16  2023                 Hairpin Ribozyme   \n",
       "17  2023              Hammerhead Ribozyme   \n",
       "18  2023                 Twister Ribozyme   \n",
       "19  2018                             tRNA   \n",
       "20  2022  Fam1A.1 aminoacylation ribozyme   \n",
       "21  2022  Fam1B.1 aminoacylation ribozyme   \n",
       "22  2022   Fam2.1 aminoacylation ribozyme   \n",
       "23  2022   Fam2.2 aminoacylation ribozyme   \n",
       "24  2022   Fam3.1 aminoacylation ribozyme   \n",
       "25  2022                        ribozyme    \n",
       "26  2016                             tRNA   \n",
       "27  2014                      GFP aptamer   \n",
       "28  2014                   NELF-E aptamer   \n",
       "29  2016                             mRNA   \n",
       "\n",
       "                               Raw Construct Sequence  \\\n",
       "0   ccccacctcttcttcttttctagAGTTGCTGCTGGGAGCTCCAGCA...   \n",
       "1   CTCTCTAGGTAGCAATATTACGACATGGAGGGAAAAGTTATCAGGC...   \n",
       "2   ATAAAAGCGCCAGAACTACAGAAATGTAGTTGACGAGGAGGAGGTT...   \n",
       "3   TAATACGACTCACTATAGGGCGCGGCATTAATGCAGCTTTATTGGA...   \n",
       "4   TAATACGACTCACTATAGGGCCGCGACTCTAGAAGTGATGCTCTGC...   \n",
       "5   TAATACGACTCACTATAGGGCCGCGACTCTAGAAGTGATGCTCTGC...   \n",
       "6   GATCTAACAAAGTTCATAAAGAAATTACTCTCGGTAGCCAAGTTGG...   \n",
       "7   TAATACGACTCACTATAGGGTCGNNNNNNNATAATCGCGTGGATAT...   \n",
       "8   GCTGTCACCGGANNNNNNNNNTCNNNTNNNTNCNNNNNNNNNTCNN...   \n",
       "9   GCTGTCACCGGANNNNNNNCNNNNTNNTGANNCCNTNNNNNNNNCG...   \n",
       "10  GCTGTCACCGGANNNNNNNCNNNNNNNNNNNGNCCNTNNNAGNNNN...   \n",
       "11  GCTGTCACCGGANNNNNNNNNNNGTCTNNNNNNNCNNNNNNNNCNN...   \n",
       "12  GCTGTCACCGGANNNNNNNNNNCGNNNTGATGNNNNNNNNNNNNCN...   \n",
       "13  GUGCCUUGCGCCGGGAAACCACGCAAGGGAUGGUGUCAAAUUCGGC...   \n",
       "14  GGGGGCCACAGCAGAAGCGTTCACGTCGCGGCCCCTGTCAGATTCT...   \n",
       "15  GGCCGGCATGGTCCCAGCCTCCTCGCTGGCGCCGGCTGGGCAACAT...   \n",
       "16  GTCCTCTTTCACGTGCGCGTGAAAGAGAGAAGTGAACCAGAGAAAC...   \n",
       "17      ACCGGATGTGCTTTCCGGTCTGATGAGTCCGTGAGGACGAAACAG   \n",
       "18   AACACTGCCAATGCCGGTCCCAAGCCCGGATAAAAGTGGAGGGGGCGG   \n",
       "19  GUUCCGUUGGCGUAAUGGUAACGCGUCUCCCUCCUAAGGAGAAGAC...   \n",
       "20  GATAATACGACTCACTATAGGGAATGGATCCACATCTACGAATTCC...   \n",
       "21  GATAATACGACTCACTATAGGGAATGGATCCACATCTACGAATTCC...   \n",
       "22  GATAATACGACTCACTATAGGGAATGGATCCACATCTACGAATTCA...   \n",
       "23  GATAATACGACTCACTATAGGGAATGGATCCACATCTACGAATTCA...   \n",
       "24  GATAATACGACTCACTATAGGGAATGGATCCACATCTACGAATTCA...   \n",
       "25  GGGGGCCACAGCAGAAGCGTTCACGTCGCGGCCCCTGTCAGATTCT...   \n",
       "26  GTTCCGTTGGCGTAATGGTAACGCGTCTCCCTCCTAAGGAGAAGAC...   \n",
       "27  GCUUCUGGACUGCGAUGGGAGCACGAAACGUCGUGGCGCAAUUGGG...   \n",
       "28  CCCAACGACUGCCGAGCGAGAUUACGCUUGAGCGCCCCACUGAGGA...   \n",
       "29  GATCCAGATCTAACTTGGGGTGGCTTTGTCTTCTTCTTTTGCCAAT...   \n",
       "\n",
       "                fitness filename  Spearman       AUC       MCC  \n",
       "0                   ke_2017_mRNA  0.383205  0.303543 -0.279279  \n",
       "1              soo_2021_ribozyme  0.009862  0.496602 -0.008671  \n",
       "2       Andreasson_2020_ribozyme  0.114690  0.447871 -0.065024  \n",
       "3        Kobori_2015_tw_ribozyme  0.018681  0.495067 -0.007264  \n",
       "4       Kobori_2015_j12_ribozyme  0.229690  0.620117  0.140625  \n",
       "5        Kobori_2015_p4_ribozyme  0.004775  0.468445 -0.046875  \n",
       "6                  Guy_2014_tRNA  0.046266  0.533315  0.064516  \n",
       "7           Kobori_2018_ribozyme  0.084779  0.459204 -0.052490  \n",
       "8     Townshend_2015_4nt_aptamer  0.093221  0.533514 -0.004854  \n",
       "9     Townshend_2015_5nt_aptamer  0.220651  0.374713 -0.200563  \n",
       "10    Townshend_2015_6nt_aptamer  0.208933  0.586520  0.113537  \n",
       "11    Townshend_2015_7nt_aptamer  0.037624  0.476610 -0.037254  \n",
       "12    Townshend_2015_8nt_aptamer  0.226219  0.385770 -0.163823  \n",
       "13            Peri_2022_ribozyme  0.026147  0.500842 -0.008057  \n",
       "14  Roberts_2023_cepeb3_ribozyme  0.219021  0.582899  0.117490  \n",
       "15     Roberts_2023_HDV_ribozyme  0.194930  0.405339 -0.130178  \n",
       "16      Roberts_2023_hp_ribozyme  0.053403  0.490231 -0.008770  \n",
       "17      Roberts_2023_hh_ribozyme  0.043122  0.474383 -0.029961  \n",
       "18      Roberts_2023_tw_ribozyme  0.019867  0.489497 -0.023699  \n",
       "19             Domingo_2018_tRNA  0.190741  0.597690  0.131018  \n",
       "20   Janzen_2022_fam1a1_ribozyme  0.082191  0.540690  0.055811  \n",
       "21   Janzen_2022_fam1b1_ribozyme  0.038526  0.531691  0.047619  \n",
       "22    Janzen_2022_fam21_ribozyme  0.264427  0.359427 -0.208398  \n",
       "23    Janzen_2022_fam22_ribozyme  0.007434  0.503460  0.008704  \n",
       "24    Janzen_2022_fam31_ribozyme  0.278336  0.345498 -0.224783  \n",
       "25            Beck_2022_ribozyme  0.206225  0.577197  0.110454  \n",
       "26                  Li_2016_tRNA  0.060912  0.529479  0.042133  \n",
       "27         Tome_2014_GFP_aptamer  0.282189  0.363751 -0.189455  \n",
       "28       Tome_2014_NELFE_aptamer  0.041492  0.457346 -0.070890  \n",
       "29              Julien_2016_mRNA  0.105890  0.447097 -0.075486  "
      ]
     },
     "execution_count": 18,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "wt_seqs"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "courtney_master",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.8.12"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
